search-provider/eks-metadata-provider-dbus.xml

```diff
@@ -0,0 +1,129 @@
+                might return multiple shards, so if the client wants to look
+                up further information for a particular content object, they
+                will need to check every shard to see if it has that content.
+       @Results: A tuple of (upper-bound, models).
+
+                 upper-bound is the number of models that would be returned
+                 in a search result if "limit" had not been applied.
+
+                 models is an array of dictionaries with metadata about each
+                 content object matched in the query. New properties may be
+                 added to these dictionaries during the course of this
+                 interface's lifetime, so callers should explicitly check if
+                 the property is present on the returned dictionary for each
+                 piece of content before attempting to use it, and if not,
+                 substitute a sensible default.
+
+                 "title": The title of the content object.
+                 "content_type": The MIME type of the underlying content data.
+                 "language": The language code of the content.
+                 "synopsis": A brief description of the content.
+                 "last_modified_date": Date (ISO8061) when the content was
+                                       last modified.
+                 "original_title": The original title this content had.
+                 "original_uri": The URI that this content was sourced from.
+                 "license": License for this content.
+                 "copyright_holder": Who holds the copyright for this content.
+                 "thumbnail_uri": An EKN ID specifying the location in one
+                                  of the returned shards where data for
+                                  the thumbnail for this content is stored.
+                 "ekn_id": An EKN ID specifying the location in one of the
+                           returned shards where data for this content is
+                           stored.
+                 "child_tags": If this content object model represents a
+                               set, the tags which should be queried as part
+                               of "tags-match-any" in Query above to find
+                               content objects inside this set.
+                 "tags": The tags for this content object model, which allow
+                         it to be associated with a set via "child_tags" or
+                         provide metadata about the type of content.
+                 "temporal-coverage": The dates this content covers.
+                 "discovery_feed_content": An opaque struct with content for
+                                           the discovery feed.
+    -->
+    <method name="Query">
+      <arg type="as" name="Shards" direction="out" />
+      <arg type="(iaa{sv})" name="Results" direction="out" />
```

Comment:
I would go for `(a{sv}aa{sv})` to allow for expansion to other `Xapian::MSet` parameters beyond just the upper bound.

Reply:
I don't really want to do that here - as specified in the doc, if we need to add additional parameters, it is necessary to bump the API version of this interface, as opposed to adding them and having them be ignored on older versions.

Reply:
Ah wait, I see this is an outparam, which does allow expansion.

Okay, I can make that change then.